src/components/Table/utils/TableState.ts

```diff
@@ -221,6 +225,25 @@ export class TableState {
     }
   }
 
+  // load and trigger column to be expanded
+  async loadExpandedColumns(column: GridColumnWithId<any>): Promise<void> {
+    // if we dont have anything in our cache and our expanded columns are a function
+    if (!this.loadedColumns.has(column.id) && isFunction(column.expandColumns)) {
+      // @ts-ignore need to type this - should be a promise
+      // set our result to the function call of expandColumns
+      const result = await column.expandColumns();
+      // once we have the loaded columns, add result to local cache
+      this.loadedColumns.set(column.id, assignDefaultColumnIds(result));
+    }
+    // once column is in local cache, then toggle
+    this.toggleExpandedColumn(column.id);
+  }
+
+  // if there is a promise, then load the expandable columns from the cache, if not then return the expandedColumns
+  getExpandedColumns(column: GridColumnWithId<any>) {
```

Comment:
**suggestion:** This is where I'm thinking we would define the response type of the function:
```suggestion
  getExpandedColumns(column: GridColumnWithId<any>): GridColumnWithId<any> {
```
Though, now I am seeing this `any` type... I wonder if that is why you needed to do the `as ...` casting above?

Reply:
yeah, typescript thinks there is an `undefined` being returned from `getExpandedColumns` because it thinks `expandedColumns` could be undefined.. but if I do this then it stops complaining

```getExpandedColumns(column: GridColumnWithId<any>): GridColumnWithId<any>[] {
    return isFunction(column.expandColumns) ? this.loadedColumns.get(column.id) ?? [] : column.expandColumns ?? [];
  }
```